api/api.proto

```diff
@@ -4450,3 +4456,57 @@ enum EnforcementType {
   // This specifies that the rule is preferred and can be best effort
   preferred = 1;
 }
+
+enum RestoreParamBoolType {
+	PARAM_BKUPSRC = 0; // Default: whateever was cloudbakup's option for the parameter
+	PARAM_FALSE = 1;
+	PARAM_TRUE = 2;
+}
+
+// RestoreSpec allows some of the restore volume properties of to be modified
+// while restoring cloud baackup.
+message RestoreVolumeSpec {
+  // HaLevel specifies the number of copies of data.
+  int64 ha_level = 1;
+  // Cos specifies the relative class of service.
+  CosType cos = 2;
+  // IoProfile provides a hint about application using this volume.
+  IoProfile io_profile = 3;
+  // SnapshotInterval in minutes, set to 0 to disable snapshots
+  uint32 snapshot_interval = 4;
+  // Shared is true if this volume can be concurrently accessed by multiple users.
+  RestoreParamBoolType shared = 5;
+  // ReplicaSet is the desired set of nodes for the volume data.
+  ReplicaSet replica_set = 6;
+  // Aggregation level Specifies the number of parts the volume can be aggregated from.
+  uint32 aggregation_level = 7;
+  // SnapshotSchedule a well known string that specifies when snapshots should be taken.
+  string snapshot_schedule = 8;
+  // Sticky volumes cannot be deleted until the flag is removed.
+  RestoreParamBoolType sticky = 9;
+  // Group identifies a consistency group
+  Group group = 10;
```

Comment:
How do you differentiate if this should be picked from the backup or not used?

Reply:
if specified, it will be used. if not backup's spec is used.

Reply:
But what if the backup spec has a group and a user doesn't want to use that for the restore?

Reply:
in that case user specifies a group in RestoreSpec. where that Group spec is enaforced or not is specified by enforce_group parameter. IOW, if group is specified in restore spec, it will be used rather than backup spec's group field.

Reply:
to clarfy, if enforce_group is set to yes, then restore may fail if the group requirements cant be met.

Reply:
Same question for ownership and storage_policy too then. BackupSpec has a storagepolicy, if we don't want to use a storage policy during restore how will a user specify that? They can't leave it empty because it'll pick up the one from the Backup

Reply:
I see.  what are you asking is ignore fields for storage policy and ownership. I will add those. enforce_group would be honored only when group is specified.   From this discussion, I see the need for ignoreGroup as well. All the ignore fields will only be honored when user does not specify the relevant fields in restore spec.

Reply:
Done. Added ignore fields for the above.

Reply:
I think this is error prone. Can those be made pointers? If it is nil then you can use the value from the Backup. If it is not nil then use the value from the variable (it can be empty or have a valid value)

Reply:
Ok. I reverted the changes. Those fields are pointers.